### ISSUE TITLE
when run test:console it eats user input

### ISSUE THREAD

**Issue by antonkulaga** [lihaoyi/Ammonite#553]
When I run ammonite as test:console it eats the letters I type. The bug appeared only in the latest version

**Comment by lihaoyi** [comment]
This is annoying me as well; anyone want to help bisect the changes and see what is causing this?

**Comment by emanresusername** [comment]
Seems like this is a scala 2.12 thing i don't see the problem in scala 2.11 projects

**Comment by emanresusername** [comment]
perhaps [this](https://issues.scala-lang.org/browse/SI-10102) is related?

**Comment by jjst** [comment]
I can confirm I'm seeing the same issue on a 2.12.1 project.

**Comment by stanch** [comment]
First happened to me with `0.8.2` on Scala `2.12.1` (`2.11.8` was fine). Now also happening with `0.8.3` on both Scala `2.11.11` and `2.12.2` :( Also this affects both `console` and `test:console` as far as I can see (which is not surprising, but I thought was worth mentioning to clarify the original post).

**Comment by lihaoyi** [comment]
What if you put it in a `main` method in your `src/test` folder and run it via `test:run` instead of `test:console`? Does that exhibit the same symptoms?

**Comment by stanch** [comment]
Yup, that works. I actually prefer that, no more `sbt reload` when I want to change the ammonite predef :)

**Comment by lihaoyi** [comment]
ok cool, must be some bad interaction with the surrounding scala console. We should make it easier for people to do the `test:run` thing then

**Comment by emanresusername** [comment]
without out having too change to much of the existing code/docs/workflows/etc, think just this substitution in the docs would do the trick?

old
```scala
initialCommands in (Test, console) := """ammonite.Main().run()"""
```
```sh
sbt projectName/test:console
```
new
```scala
sourceGenerators in Test += Def.task {
  val file = (sourceManaged in Test).value / "amm.scala"
  IO.write(file, """object amm extends App { ammonite.Main().run() }""")
  Seq(file)
}.taskValue
```
```sh
sbt projectName/test:run
``` 
or if there are other main methods in the `Test` scope
```sh
sbt projectName/test:run-main amm
```

seems to be working for me, but i haven't tested exhaustively

**Comment by lihaoyi** [comment]
@emanresusername that should work. Could you send a PR?

**Comment by lihaoyi** [comment]
Closed by #596
